### ISSUE TITLE
Stateful model

### ISSUE THREAD

**Issue by devakar** [hualiu01/LSTMRNNStockr#1]
lstm.py create different types of model. But all these models are stateless, that means the cell state of LSTM gets reset after every batch of input. Is it taking advantage of the capability of LSTM to learn long-term dependencies?

**Comment by hualiu01** [comment]
Hi Devakar,

It's a good question. I overlooked this factor. To make a model
stateful you can specify 'shuffle=False' when calling model.fit() function
in file predict.py. And the default setting is indeed 'shuffle=True',
which means stateless. You can experiment with this on your own. I'd like
to hear that this will improve the prediction accuracy.

However, Tensorflow uses a truncated version of backpropagation which I
mentioned in the video and report(fig 3.3). And this is set by the
'normalise_window' parameter when doing data windowing. So the ideally
"infinite " long-term dependency can never be achieved in practice.

Good luck,
Hua


On Mar 12, 2018, at 7:40 AM, Devakar Kumar Verma <notifications@github.com>
wrote:

lstm.py create different types of model. But all these models are
stateless, that means the cell state of LSTM gets reset after every batch
of input. Is it taking advantage of the capability of LSTM to learn
long-term dependencies?

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/hualiu01/LSTMRNNStockr/issues/1>, or mute the thread
